### DISCUSSION TITLE
Preliminary GitHub/Prject-management for SALURBAL Climate

### DISCUSSION THREAD

**Discussion by ran-codes** [Drexel-UHC/data-science#40] [Meeting Agenda/Notes]
Attedants: @gotom22 @juruchi @ran-codes 

# Rought notes

### Agenda Draft

- how has GH been used so far in SALURBAL? (existing repos, purposes, ..)
- how we think we could/should use GH for SALURBAL-C (purposes, i.e. project management vs. “contents”/data/code/docs, etc.)
- brainstorm towards a proposal to EC (?) on: 
- how to set up GH (repos, purposes, roles, admin/access rights, …)
- complementary structures (Gdrive, data server/portal,  etc.)
- GH vs. non-GH intersection (some understanding of work flows, info flows between GH users and non-users, in particular wrt senior collaborators/EC etc.)
### SALURBAL Climate Aims

- Aim 1: Data collection + Analysis
- Aim 2: Translation research aka go scientific evidence to products
	- take general truths established in Aim 1 and apply to a local context a local case study
	- e.g. the air pollution in your population is causing ... deaths in... the next years. 
	- 2.1: very local to Bogata and ... Chile
	- 2.2: is more generic tool/method case study compilation streamline pipeline to 
- Aim 3: Research Capacity building
- Aim 4: Applied/real-world 

### Overflow of ideas

- a lot of ideas being thrown around for Aim 1 and Aim 2
- it would have a great place to compile then plana  work flow to triage and implement if high priority (As you said @usamabilal )

### Discussion

- write a white paper on potential applications of GitHub for SALURBAL-climate/p20
- Thomas more focused on
	- Aim 1 (CEC)
	- Aim 2.2 ()
		- dedicated repository for aim 2.2 
			- issues
			- meeting notes/discussions
- Jess
	- what are the types of issues to come up
	- who are the users?
		- country liaisons?
		- but what happens when the liaison is not on GitHub. 
- key question:
	- how does information exchange happen before GitHub users to the outside work
		- outside > e.g executives or higher ups
- Next steps:
	- longer goal: pitch this to executive community
	- short term get a POC
		- which GitHUb Org?
		- how many repos?

**Comment by ran-codes** [comment]
### interesting snip from meeting

![image](https://github.com/Drexel-UHC/data-science/assets/45013044/c8e49e1f-45f7-4194-8764-f2b225220985)

(need to really fill out the P20 stuff haha)!

**Comment by gotom22** [comment]
I started brainstorming [here](https://docs.google.com/presentation/d/175McrMezQ4wuTUJbSkQ--gAg9kORBLt7dJ34fl94os8/edit#slide=id.p). 

I will need to educate myself on existing repos and role of DMC still. TBC. 

![image](https://github.com/Drexel-UHC/data-science/assets/776487/7da8bfb7-ea42-48a7-b136-b2004a084e3b)

**Reply by gotom22** [reply]
@ran-codes @juruchi 
I'm already backtracking from my initial vision of this GH setup process. 

- [ ] Rather than first figuring out a comprehensive org/repo structure, I will go ahead and set up repos for CEC and 2.2 already now. The demands there are already too overwhelming to postpone this much longer. 

- [ ] Whenever we feel ready for it, we can then move these (and others) repos into a joint organization. This could be Drexel-UHC, or cleaner but more ambitious a new org (e.g. SALURB - incl. P20, SALURBAL, SALURBAL-C,  or just SALURBAL-C). I think to weigh pro's and con's upfront is not trivial, so I think we should not rush re-organization-ing. 

Let me know if you have any concerns with this approach (especially, would it set things in stone which we may want to change later?)

**Reply by ran-codes** [reply]
@gotom22 Agree with your thoughts. I think the important thing is teams start tracking on GitHub ... ASAP. we can worry about the 'GitHub organization' problem later. decentralized repos are fine ... in fact its probably better that way... as each team can organically figure out how they want to use it before we set anything in stone as a collective.

That being said can you invite @ran-codes @juruchi and @usamabilal to the repos you create? So we can all stay in touch and collaborate during this initial phase

**Reply by gotom22** [reply]
So basically, I'm thinking of the repo's function in the following hierarchy:

1. primarily as a "tool for the project managers of the various cores", and then 
2. secondarily as as "collaboration tool for the collaborators within a core", and 
3. lastly, as a "collaboration tool across cores"

Ideally we get buy-in for all 3 purposes, but whether we get this or not will determine which repos we ultimately deal with. 

Step 1 I guess is which "core project managers" will opt-in to use GH for point 1. 
etc.

**Reply by usamabilal** [reply]
I'll know more after i hear the whole thing, but the key is to have buy-in from VERY specific actors (the "core project managers"). For example for Aim 1 DMC that's me and Kari (especially Kari!). Ran, we should also work on ensuring that Goro understands the github process well (Jess, you can help us with that too), as I have seen some hesitance. Otherwise we'll have the vital registration part separate which wouldnt make sense.

**Reply by gotom22** [reply]
@usamabilal fully agree. Buy-in is what will make this work. 
That said, I imagine this (i.e. "the SALURBAL GitHub environment"(?)) to grow in a modular, organic way. Minimal buy-in (e.g. .. just me ...) would mean limited use for some project management tasks (..basically me just keeping my tasks in order...). Buy-in from collaborators will allow us to use some of the collaborating features. Buy-in from multiple aims/cores ("core project managers", and eventually also their collaborators) would allow us to streamline some of these "cross-aims/cores" processes/collaborations. Etc. 
So, in a way, "buy-in from others" is not a crucial pre-requisite from the start. 
That said, and this will remain true throughout, we will always have to make sure that GH does not result in "exclusion". So basically, there is a requirement to communicate/translate to those not using GH (to me that means, key documents (call agendas, summaries, key decisions, briefing supervisors, emails, etc.). In an ideal world, we can take care of "the noise" in GH, and only "good substance" is passed through the other communication channels...

I think with DMC, CEC and Aim2.2 we'll have a good number of "inoculation points". I suggest let's see how it goes and grows from there.
